lib/cocoapods/command/repo/add.rb

```diff
@@ -38,12 +38,12 @@ def run
           prefix = @shallow ? 'Creating shallow clone of' : 'Cloning'
           UI.section("#{prefix} spec repo `#{@name}` from `#{@url}`#{" (branch `#{@branch}`)" if @branch}") do
             config.repos_dir.mkpath
-            Dir.chdir(config.repos_dir) do
+            Pod.chdir(config.repos_dir) do
```

Comment:
I don't think this chdir is necessary, git checkout takes a directory to check out as it's last argument. Let's just make this the absolute path, something like:

```ruby
command = "clone '#{@url}' #{config.repos_dir + @name}"
```